sw/device/silicon_creator/manuf/base/ft_personalize.c

```diff
@@ -405,12 +460,162 @@ static status_t personalize_gen_dice_certificates(ujson_t *uj) {
       (hmac_digest_t *)certgen_inputs.owner_measurement,
       (hmac_digest_t *)certgen_inputs.owner_manifest_measurement,
       certgen_inputs.owner_security_version, &cdi_1_key_ids, &curr_pubkey,
-      &tbs_certs.certs[tbs_certs.next_free], &curr_cert_size));
-  tbs_certs.next_free += curr_cert_size;
-  LOG_INFO("Generated CDI_1 certificate.");
+      all_certs, &curr_cert_size));
+  cdi_1_offset = perso_blob_to_host.next_free;
+  return prepare_cert_for_shipping("CDI_1", false, all_certs, curr_cert_size);
+}
+
+/**
+ * A helper structure for quick access to a certificate stored as a perso LTV
+ * object.
+ **/
+typedef struct cert_block {
+  size_t obj_size;
+  size_t wrapped_cert_size;
+  const void *wrapped_cert_p;
+  char name[kCrthNameSizeFieldMask + 1];
+} cert_block_t;
+
+/**
+ *  Given the pointer to an LTV object, in case this is an endorsed certificate
+ *  set up the cert_block_t structure for it.
```

Comment:
nit: ```suggestion
 * Given the pointer to an LTV object, in case this is an endorsed certificate
 * set up the cert_block_t structure for it.
```

Reply:
I might have missed this one, it's not clear what's suggested here.